src/ml/neural_net/style_transfer/mps_style_transfer.m

```diff
@@ -0,0 +1,419 @@
+                                             nodeHandler: nil];
+
+    _trainingGraph = [MPSNNGraph graphWithDevice:_dev
+                                    resultImages:@[lastNodes[0].resultImage, lastNodes[1].resultImage]
+                                resultsAreNeeded:&resultsAreNeeded[0]];
+
+    _inferenceGraph = [MPSNNGraph graphWithDevice:_dev
+                                      resultImage:_model.forwardPass
+                              resultImageIsNeeded:YES];
+
+    _trainingGraph.format = MPSImageFeatureChannelFormatFloat32;
+    _inferenceGraph.format = MPSImageFeatureChannelFormatFloat32;
+  }
+  return self;
+}
+ 
+- (NSDictionary<NSString *, NSData *> *)exportWeights {
+  return [_model exportWeights];
+}
+
+- (NSDictionary<NSString *, NSData *> *)predict:(NSDictionary<NSString *, NSData *> *)inputs {
+  NSUInteger contentWidth;
+  NSUInteger contentHeight;
+  NSUInteger contentChannels;
+
+  [inputs[@"imageWidth"] getBytes:&contentWidth length:sizeof(contentWidth)];
+  [inputs[@"imageHeight"] getBytes:&contentHeight length:sizeof(contentHeight)];
+  [inputs[@"imageChannels"] getBytes:&contentChannels length:sizeof(contentChannels)];
+
+  MPSImageDescriptor *imgDesc = [MPSImageDescriptor
+    imageDescriptorWithChannelFormat:MPSImageFeatureChannelFormatFloat32
+                               width:contentWidth
+                              height:contentHeight
+                     featureChannels:contentChannels
+                      numberOfImages:1
+                               usage:MTLTextureUsageShaderWrite | MTLTextureUsageShaderRead];
+
+  NSMutableArray<MPSImage *> *contentImageArray;
+
+  for (NSUInteger index = 0; index < _batchSize; index++) {
+    NSString* key = [NSString stringWithFormat:@"%@%lu", @"contentImage", index];
+    MPSImage *contentImage = [[MPSImage alloc] initWithDevice:_dev imageDescriptor:imgDesc];
+    [contentImage writeBytes:inputs[key].bytes dataLayout:(MPSDataLayoutHeightxWidthxFeatureChannels)imageIndex:0];
+    [contentImageArray addObject:contentImage];
+  }
+
+  MPSImageBatch *contentImageBatch = [contentImageArray copy];  
+
+  id<MTLCommandBuffer> cb = [_commandQueue commandBuffer];
+
+  NSMutableArray *intermediateImages = [[NSMutableArray alloc] initWithArray:@[]];
+  NSMutableArray *destinationStates = [[NSMutableArray alloc] initWithArray:@[]];
+
+  MPSImageBatch *stylizedImages =  [_inferenceGraph encodeBatchToCommandBuffer:cb
+                                                       sourceImages:@[contentImageBatch]
+                                                       sourceStates:nil
+                                                 intermediateImages:intermediateImages
+                                                  destinationStates:destinationStates];
+
+  for (MPSImage *image in stylizedImages) {
+    [image synchronizeOnCommandBuffer:cb];  
+  }
+
+  [cb commit];
+  [cb waitUntilCompleted];
```

Comment:
This is okay for now, but in the long term we'll want to allow this to be asynchronous, so that we can start preparing the next mini-batch on CPU while this one executes on GPU. See for example the code in `mps_graph_cnnmodule::predict`

Reply:
Added a todo for a future PR.